File: 35-SearchInsertPosition/main.py
from typing import List

class Solution:
  def search_insert(self, nums: List[int], target: int) -> int:
    left, right = 0, len(nums) - 1

    while left <= right:
      mid = (left + right) // 2

      if nums[mid] == target:
        return mid 
      elif nums[mid] < target:
        left = mid + 1 
      else:
        right = mid - 1

    return left 

s = Solution()
print(s.search_insert([1, 3, 5, 6], 5))
print(s.search_insert([1, 3, 5, 6], 2))
print(s.search_insert([1, 3, 5, 6], 7))

File: 35-SearchInsertPosition/main.ts
const searchInsert = (nums: number[], target: number): number => {
  let left: number = 0;
  let right: number = nums.length - 1;

  while (left <= right) {
    const mid: number = Math.floor((left + right) / 2);

    if (nums[mid] === target) {
      return mid;
    } else if (nums[mid] <= target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return left;
}

console.log(searchInsert([1, 3, 5, 6], 5));
console.log(searchInsert([1, 3, 5, 6], 2));
console.log(searchInsert([1, 3, 5, 6], 7));